spec/lib/extensions/ar_recent_records_spec.rb

```diff
@@ -0,0 +1,50 @@
+describe "AR Count By Date extension" do
+  context "vms with dates" do
+    before do
+      FactoryGirl.create(:vm, :ems_id => 1, :created_on => 0.days.ago)
```

Comment:
Minor, but you could potentially have a sporadic test failure here if the test *happens* to run at exactly midnight.  Because each of these `.days.ago` is running a separately Time calculation. To prevent that you can set up the test to freeze time around it..

```ruby
around do |example|
  Timecop.freeze { example.call }
end
```

Reply:
Ok, added it, though I'm not entirely sure I did it properly.